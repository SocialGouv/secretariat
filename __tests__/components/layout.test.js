import { render } from "@testing-library/react"
import { useSession } from "next-auth/react"

import Layout from "../../src/components/layout"

jest.mock("next-auth/react", () => ({
  useSession: jest.fn(),
}))

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
})

it("renders layout", () => {
  useSession.mockReturnValueOnce([false, false])

  const { container } = render(
    <Layout>
      <div>some children here</div>
    </Layout>
  )

  expect(container).toMatchSnapshot()
})
