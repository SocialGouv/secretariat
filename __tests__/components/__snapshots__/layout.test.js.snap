// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders layout 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="header-content"
    >
      <a
        href="/"
      >
        <div
          class="logo"
        >
          République
          <br />
          Française
        </div>
        <div
          class="title"
        >
          <h1>
            Secretariat
          </h1>
          <p>
            Gestion des comptes de services @SocialGouv
          </p>
        </div>
      </a>
      <nav />
    </div>
  </header>
  <div
    class="page"
  >
    <div>
      some children here
    </div>
    <footer>
      <div
        class="footer-content"
      >
        <div
          class="logo-info"
        >
          <div
            class="logo big"
          >
            République
            <br />
            Française
          </div>
          <div
            class="info"
          >
            <p>
              <strong>
                Secretariat
              </strong>
               est une application permettant de gérer les accès des utilisateurs aux services de la Fabrique Numérique.
            </p>
            <p>
              Pour en savoir plus sur l'utilisation de
               
              <strong>
                Secretariat
              </strong>
              , visitez le
               
              <a
                href="https://github.com/SocialGouv/www/wiki/Inscrire-son-%C3%A9quipe-au-standup"
                rel="noopener noreferrer"
                target="_blank"
              >
                wiki
              </a>
              .
            </p>
            <p
              class="flex items-center"
            >
              <i
                class="ri-github-fill ri-xl relative top-px mr-1"
              />
              version 
               (
              <a
                href="https://github.com/SocialGouv/secretariat/tree/undefined"
                rel="noreferrer"
                target="_blank"
              />
              )
            </p>
          </div>
        </div>
        <div
          class="links"
        >
          <a
            href="https://www.fabrique.social.gouv.fr/"
            rel="noreferrer"
            target="_blank"
          >
            La Fabrique Numérique
          </a>
          <a
            href="/legal"
          >
            Mentions légales
          </a>
          <button>
            <i
              class="theme-toggle-button ri-sun-line"
            />
          </button>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
